clients/java/src/test/java/io/camunda/zeebe/client/ZeebeClientTest.java

```diff
@@ -728,6 +738,125 @@ public void shouldNotSetDefaultJobWorkerTenantIdsFromSetterWithCloudClientBuilde
         .isEqualTo(builder);
   }
 
+  @Test
+  @DisplayName("should throw when gatewayTarget is invalid")
+  public void shouldThrowWhenGatewayTargetIsInvalid() {
+    // given
+    final ZeebeClientBuilderImpl builder = new ZeebeClientBuilderImpl();
+
+    // when
+    Assertions.assertThrows(RuntimeException.class, () -> builder.gatewayTarget("0.0.0.0:26500"));
+  }
+
+  @Test
+  @DisplayName("should use gatewayAddress when gatewayAddress is valid ip:port")
+  public void shouldUseGatewayAddressWhenGatewayAddressIsValidIpPort() {
+    // given
+    final ZeebeClientBuilderImpl builder = new ZeebeClientBuilderImpl();
+    final Properties properties = new Properties();
+    properties.putIfAbsent(ClientProperties.GATEWAY_ADDRESS, "0.0.0.0:26500");
+    builder.withProperties(properties);
+
+    // when
+    builder.build();
+
+    // then
+    Assertions.assertNull(builder.getGatewayTarget());
+
+    Assertions.assertEquals("0.0.0.0:26500", builder.getGatewayAddress());
+  }
+
+  @Test
+  @DisplayName("should use gatewayTarget when gatewayAddress is not valid ip:port")
+  public void shouldUseGatewayTargetWhenGatewayAddressIsNotValidIpPort() {
+    // given
+    final ZeebeClientBuilderImpl builder = new ZeebeClientBuilderImpl();
+    final Properties properties = new Properties();
+    properties.putIfAbsent(ClientProperties.GATEWAY_ADDRESS, "gateway.zeebe.com:26500");
+
+    // use
+    builder.withProperties(properties);
+
+    // verify
+    Assertions.assertEquals("gateway.zeebe.com:26500", builder.getGatewayTarget());
+  }
+
+  @Test
+  @DisplayName("should use gatewayAddress to buildChannel when gatewayTarget is null")
+  public void shouldUseGatewayAddressToBuildChannelWhenGatewayTargetIsNull() {
+
+    // given
+    final ZeebeClientConfiguration zeebeClientConfiguration =
+        mock(ZeebeClientConfiguration.class, RETURNS_DEEP_STUBS);
+    when(zeebeClientConfiguration.getGatewayTarget()).thenReturn(null);
+    when(zeebeClientConfiguration.getKeepAlive().toMillis()).thenReturn(1L);
+    when(zeebeClientConfiguration.getInterceptors()).thenReturn(new ArrayList<>());
+    when(zeebeClientConfiguration.getGatewayAddress()).thenReturn("0.0.0.0:26500");
+
+    // use & verify
+    Assertions.assertDoesNotThrow(
+        () -> {
+          client = new ZeebeClientImpl(zeebeClientConfiguration);
+        });
+
+    // verify
+    verify(zeebeClientConfiguration, times(1)).getGatewayAddress();
+  }
+
+  @Test
+  @DisplayName("should use gatewayTarget to buildChannel when gatewayTarget is null")
+  public void shouldUseGatewayTargetToBuildChannelWhenGatewayTargetIsNull() {
```

Comment:
:x: I assume you meant to write **NotNull**:
```suggestion
  @DisplayName("should use gatewayTarget to buildChannel when gatewayTarget is not null")
  public void shouldUseGatewayTargetToBuildChannelWhenGatewayTargetIsNotNull() {
```

Reply:
Thanks for your review❤! I will revise these as you suggest when I have time.